test/jdk/java/awt/Frame/MultiScreenTest.java

```diff
@@ -0,0 +1,484 @@
+/*
+ * Copyright (c) 2000, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Canvas;
+import java.awt.Choice;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.FontMetrics;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Image;
+import java.awt.Label;
+import java.awt.LayoutManager;
+import java.awt.Panel;
+import java.awt.Rectangle;
+import java.awt.RenderingHints;
+import java.awt.TextField;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import java.awt.image.ColorModel;
+import java.awt.image.MemoryImageSource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JFrame;
+
+import jtreg.SkippedException;
+
+/*
+ * @test
+ * @bug 4312921
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame
+ * @summary Tests that no garbage is painted on primary screen with DGA
+ * @run main/manual MultiScreenTest
+ */
+
+public class MultiScreenTest {
+    static GraphicsEnvironment ge;
+    static GraphicsDevice[] gs;
+
+    public static void main(String[] args) throws Exception {
+        ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+        gs = ge.getScreenDevices();
+        if (gs.length < 2) {
+            throw new SkippedException("You have only one monitor in your system - test passed");
+        }
+        MultiScreenTest obj = new MultiScreenTest();
```

Comment:
Although this test will bail if there aren't two monitors, I'd like to see us remember to add @key multimon to such tests.
Then if someone wants to run multimon testsand ONLY those tests they can specify this key 
I'm sure we missed a couple of opportunities to do it already but let's not miss this one.

See 
https://github.com/openjdk/jdk/blob/master/test/jdk/TEST.ROOT

Reply:
Ok.. added ` @key multimon `.